### ISSUE TITLE
No access to any values with dash in the name

### ISSUE THREAD

**Issue by moabu** [rancher/partner-charts#235]
🐛 🐛 🐛  
#### Steps to reproduce 

1. Create a `questions.yaml` with the following

```yaml
# MyExampleApp
- variable: global.myapp-api.enabled
  default: true
  type: boolean
  group: "Optional Services"
  required: true
  label: Enable MyAppAPI
  description: "What a cool app!"
- variable: myapi-app.replicas
  default: 1
  required: false
  type: int
  group: "Replicas"
  label: MyAppAPI Replicas
  description: "Service replica number."
  show_if: "global.myapp-api.enabled=true"
```
  
2. Follow [instructions](https://github.com/rancher/partner-charts/tree/main-source) to generate and test the chart . Add the repository to the UI.

3. Inside the UI, head to the Charts section and click on the chart added. Choose the option enabling the example app. 

4. The Replicas group does not appear. 

This happens with all dashed keys.

#### Current behaviour (bug)

It seems like any key that contains a dash `-` in the `questions.yaml` is not accessed, and hence does not resolve correctly in the UI. 

```yaml
# MyExampleApp
- variable: global.myapp-api.enabled
  default: true
  type: boolean
  group: "Optional Services"
  required: true
  label: Enable MyAppAPI
  description: "What a cool app!"
- variable: myapi-app.replicas
  default: 1
  required: false
  type: int
  group: "Replicas"
  label: MyAppAPI Replicas
  description: "Service replica number."
  show_if: "global.myapp-api.enabled=true"
```

#### Expected behaviour (correct)

I expect the UI `show_if`  and `show_subquestion_if` to respond to dashed keys.

#### Relevant Rancher logs

```
HEAD is now at 8a66d84 test:re
2021-11-04 08:36:45.202767 I | mvcc: store.index: compact 41063
2021-11-04 08:36:45.240412 I | mvcc: finished scheduled compaction at 41063 (took 36.708926ms)
2021/11/04 08:36:45 [TRACE] REST LIST api//v1//componentstatuses
2021/11/04 08:37:00 [TRACE] REST LIST api//v1//componentstatuses
2021/11/04 08:37:15 [TRACE] REST LIST api//v1//componentstatuses
HEAD is now at cc5a839 Merge pull request #228 from alex-isv/main
HEAD is now at cc5a839 Merge pull request #228 from alex-isv/main
HEAD is now at 1a6477e Bump harvester csi driver 0.1.4
HEAD is now at 1a6477e Bump harvester csi driver 0.1.4
HEAD is now at 1a6477e Bump harvester csi driver 0.1.4
HEAD is now at cc5a839 Merge pull request #228 from alex-isv/main
2021/11/04 08:37:24 [DEBUG] [etcd-backup] checking backups for cluster [local]
2021/11/04 08:37:24 [DEBUG] [etcd-backup] [local] is not an rke cluster, skipping..
2021/11/04 08:37:30 [TRACE] REST LIST api//v1//componentstatuses
HEAD is now at 7bd8cb6 Merge pull request #1577 from thedadams/release-v2.6-2.5.10-forward-port
2021/11/04 08:37:32 [DEBUG] Skipping installing/upgrading desired version 100.0.1+up0.3.7 for release fleet, current version 100.0.1+up0.3.7 is greater or equal to minimal required version 100.0.1+up0.3.7
HEAD is now at 7bd8cb6 Merge pull request #1577 from thedadams/release-v2.6-2.5.10-forward-port
2021/11/04 08:37:32 [DEBUG] Skipping installing/upgrading desired version 100.0.1+up0.3.7 for release fleet-crd, current version 100.0.1+up0.3.7 is greater or equal to minimal required version 100.0.1+up0.3.7
2021/11/04 08:37:32 [DEBUG] Skipping installing/upgrading desired version 1.0.1+up0.2.1 for release rancher-webhook, current version 1.0.1+up0.2.1 is greater or equal to minimal required version 1.0.1+up0.2.1
2021/11/04 08:37:33 [TRACE] SettingController calling handler mgmt-auth-settings-controller server-url
2021/11/04 08:37:33 [TRACE] SettingController calling handler rke-metadata-handler server-url
2021/11/04 08:37:33 [DEBUG] DesiredSet - No change(2) /v1, Kind=ServiceAccount cattle-system/git-webhook-api-service for apiservice git-webhook
2021/11/04 08:37:33 [DEBUG] DesiredSet - No change(2) /v1, Kind=Secret cattle-fleet-system/stv-aggregation for apiservice git-webhook
HEAD is now at 7bd8cb6 Merge pull request #1577 from thedadams/release-v2.6-2.5.10-forward-port
2021/11/04 08:37:33 [TRACE] SettingController calling handler mgmt-auth-settings-controller server-url
2021/11/04 08:37:33 [TRACE] SettingController calling handler rke-metadata-handler server-url
2021/11/04 08:37:33 [DEBUG] DesiredSet - No change(2) /v1, Kind=ServiceAccount cattle-system/git-webhook-api-service for apiservice git-webhook
2021/11/04 08:37:33 [DEBUG] DesiredSet - No change(2) /v1, Kind=Secret cattle-fleet-system/stv-aggregation for apiservice git-webhook
2021/11/04 08:37:33 [DEBUG] Skipping installing/upgrading desired version 100.0.1+up0.3.7 for release fleet, current version 100.0.1+up0.3.7 is greater or equal to minimal required version 100.0.1+up0.3.7
2021/11/04 08:37:33 [DEBUG] Skipping installing/upgrading desired version 100.0.1+up0.3.7 for release fleet-crd, current version 100.0.1+up0.3.7 is greater or equal to minimal required version 100.0.1+up0.3.7
2021/11/04 08:37:33 [DEBUG] Skipping installing/upgrading desired version 1.0.1+up0.2.1 for release rancher-webhook, current version 1.0.1+up0.2.1 is greater or equal to minimal required version 1.0.1+up0.2.1
2021/11/04 08:37:45 [TRACE] REST LIST api//v1//componentstatuses
```

#### Reported by
@moabu

🐛 🐛 🐛

**Comment by recena** [comment]
@nflondo Could you work on this?

**Comment by nflondo** [comment]
I wasn't able to duplicate this issue with Rancher server 2.7.4.  I created a [test Helm chart](https://github.com/nflondo/helm-repo/tree/main/charts/nginxtest) and a `questions.yaml `as indicated in this issue and was able to see `Replicas` group when selectidng `Enable MyAppAPI` as shown in picture below:
![helm-install-questions](https://github.com/rancher/partner-charts/assets/11321470/15021e9b-0130-4ebc-8bb9-1d576e3522c7)
Also, it seems like application `Gluu Cloud Identity and Access Management` for which this issue might have been opened, is correctly displaying `Replicas` group during install in the UI:
![gluu-app](https://github.com/rancher/partner-charts/assets/11321470/8314fe17-796d-4094-8fed-eb9532889c04)

It seems this issue has been fixed since it was opened.  Closing.
